src/engine/naive_engine.cc

```diff
@@ -55,20 +69,41 @@ class NaiveEngine final : public Engine {
     opr->opr_name = opr_name;
     return opr;
   }
+
   void DeleteOperator(OprHandle op) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
     delete opr;
   }
+
   void Push(OprHandle op, Context exec_ctx, int priority = 0, bool profiling = false) override {
```

Comment:
it looks like Push is never called in NaiveEngine. Have you tested this? what's calling Push for naiveengine?